### PR TITLE
Adding additional documentation

### PR DESCRIPTION
**Pull request by noxora**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

**Description of change**

**Which issue this PR fixes**

### CODE REVIEW

File: doc/plugin_server_nodeattestor_x509pop.md [hunk 1]
```diff
@@ -41,3 +41,15 @@ A sample configuration:
...
+The agent path template is a way of customizing the format of generated SPIFFE IDs for agents.
+The template formatter is using Golang text/template conventions, it can reference values provided by the plugin or in a golang x509.Certificate
+
+The some useful values are:
+
+- .PluginName
+- .Fingerprint
+- .TrustDomain
+- .Subject.CommonName
+- .Subject.SerialNumber
```
Comment (azdagron) [inline doc/plugin_server_nodeattestor_x509pop.md:55]: I wonder if this wouldn't be best represented in a table with a column for a description? Some of these are straightforward but others might need a little explanation.

File: doc/plugin_server_nodeattestor_x509pop.md [hunk 2]
```diff
@@ -41,3 +41,15 @@ A sample configuration:
 | ------------------- | --------------------------------------------------------- | --------------------------------------------------------------------- |
 | Common Name         | `x509pop:subject:cn:example.org`                                  | The Subject's Common Name (see X.500 Distinguished Names)             |
 | SHA1 Fingerprint    | `x509pop:ca:fingerprint:0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33` | The SHA1 fingerprint as a hex string for each cert in the PoP chain, excluding the leaf.  |
+
+## Agent Path Template
+The agent path template is a way of customizing the format of generated SPIFFE IDs for agents.
+The template formatter is using Golang text/template conventions, it can reference values provided by the plugin or in a golang x509.Certificate
+
+The some useful values are:
```
Comment (azdagron) [inline doc/plugin_server_nodeattestor_x509pop.md:49]:
```suggestion
Some useful values are:
```

File: doc/plugin_server_nodeattestor_x509pop.md [hunk 3]
```diff
@@ -41,3 +41,15 @@ A sample configuration:
 | ------------------- | --------------------------------------------------------- | --------------------------------------------------------------------- |
 | Common Name         | `x509pop:subject:cn:example.org`                                  | The Subject's Common Name (see X.500 Distinguished Names)             |
 | SHA1 Fingerprint    | `x509pop:ca:fingerprint:0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33` | The SHA1 fingerprint as a hex string for each cert in the PoP chain, excluding the leaf.  |
+
+## Agent Path Template
+The agent path template is a way of customizing the format of generated SPIFFE IDs for agents.
+The template formatter is using Golang text/template conventions, it can reference values provided by the plugin or in a golang x509.Certificate
```
Comment (azdagron) [inline doc/plugin_server_nodeattestor_x509pop.md:47]: Maybe change this to a link to the pkg.go.dev documentation?

https://pkg.go.dev/crypto/x509#Certificate

**Review by azdagron — COMMENTED** [review]
Hey thanks for opening this @noxora ! Any efforts to improve the documentation are appreciated! Just a few comments...

**Comment by noxora** [comment]
I've gone ahead and made the requested changes, let me know if you'd like commits squashed or any further changes 😄

**Review by azdagron — APPROVED** [review]
LGTM! Thanks, @noxora !

**Pull request merged**